src/argilla/tasks/async_typer.py

```diff
@@ -0,0 +1,51 @@
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import asyncio
+import sys
+from functools import wraps
+from typing import Any, Callable, Coroutine, TypeVar
+
+import typer
+
+if sys.version_info < (3, 10):
+    from typing_extensions import ParamSpec
+else:
+    from typing import ParamSpec
+
+
+P = ParamSpec("P")
+R = TypeVar("R")
+
+
+# https://github.com/tiangolo/typer/issues/88#issuecomment-1613013597
+class AsyncTyper(typer.Typer):
+    def async_command(
+        self, *args: Any, **kwargs: Any
+    ) -> Callable[[Callable[P, Coroutine[Any, Any, R]]], Callable[P, Coroutine[Any, Any, R]]]:
+        def decorator(async_func: Callable[P, Coroutine[Any, Any, R]]) -> Callable[P, Coroutine[Any, Any, R]]:
+            @wraps(async_func)
+            def sync_func(*_args: P.args, **_kwargs: P.kwargs) -> R:
+                return asyncio.run(async_func(*_args, **_kwargs))
+
+            self.command(*args, **kwargs)(sync_func)
+            return async_func
+
+        return decorator
+
+
+def run(function: Callable[..., Coroutine[Any, Any, Any]]) -> None:
```

Comment:
I think using this [alternative](https://github.com/tiangolo/typer/issues/88#issuecomment-1616045478) (already in the same thread) we don't need the `run` function and adding the extra `async_typer.run(command)` line on each command definition

Reply:
I've update the `AsyncTyper` class and now `Typer.command` functionality is extended, but we still need the `run` function as we need to use the `AsyncTyper.command` method (which is the one able to execute both sync and async functions).